kv/kvserver: TestLargeUnsplittableRangeReplicate failed

kv/kvserver.TestLargeUnsplittableRangeReplicate [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/12275044?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/12275044?buildTab=artifacts#/) on master @ [72c01adec22d81a18b855b66648b3f79a2990101](https://github.com/cockroachdb/cockroach/commits/72c01adec22d81a18b855b66648b3f79a2990101):


```
=== RUN   TestLargeUnsplittableRangeReplicate
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/be2c2653728c52ac0d08c8972ba7c01d/logTestLargeUnsplittableRangeReplicate644991607
    test_log_scope.go:81: use -show-logs to present logs inline
    replicate_queue_test.go:1869: replicas: {1,2,3}
    replicate_queue_test.go:1869: replicas: {1,2,3}
    replicate_queue_test.go:1869: replicas: {1,2,3}
    replicate_queue_test.go:1869: replicas: {1,2,3}
    replicate_queue_test.go:1869: replicas: {1,2,3}
    replicate_queue_test.go:1869: replicas: {1,2,3}
    replicate_queue_test.go:1869: replicas: {1,2,3}
    replicate_queue_test.go:1869: replicas: {1,2,3}
    replicate_queue_test.go:1869: replicas: {1,2,3}
    replicate_queue_test.go:1869: replicas: {1,2,3}
    replicate_queue_test.go:1869: replicas: {1,2,3}
    replicate_queue_test.go:1869: replicas: {1,2,3}
    replicate_queue_test.go:1869: replicas: {1,2,3}
    replicate_queue_test.go:1869: replicas: {1,2,3}
    replicate_queue_test.go:1869: replicas: {1,2,3}
    replicate_queue_test.go:1869: replicas: {1,2,3}
    replicate_queue_test.go:1869: replicas: {1,2,3}
    replicate_queue_test.go:1869: replicas: {1,2,3}
    replicate_queue_test.go:1869: replicas: {1,2,3}
    replicate_queue_test.go:1869: replicas: {1,2,3}
    replicate_queue_test.go:1869: replicas: {1,2,3}
    replicate_queue_test.go:1869: replicas: {1,2,3}
    replicate_queue_test.go:1869: replicas: {1,2,3}
    replicate_queue_test.go:1869: replicas: {1,2,3}
    replicate_queue_test.go:1869: replicas: {1,2,3}
    replicate_queue_test.go:1869: replicas: {1,2,3}
    replicate_queue_test.go:1869: replicas: {1,2,3}
    replicate_queue_test.go:1869: replicas: {1,2,3}
    replicate_queue_test.go:1869: replicas: {1,2,3}
    replicate_queue_test.go:1869: replicas: {1,2,3}
    replicate_queue_test.go:1869: replicas: {1,2,3}
    replicate_queue_test.go:1869: replicas: {1,2,3}
    replicate_queue_test.go:1869: replicas: {1,2,3}
    replicate_queue_test.go:1869: replicas: {1,2,3}
    replicate_queue_test.go:1869: replicas: {1,2,3,4,5}
    replicate_queue_test.go:1877: condition failed to evaluate within 45s: sql: no rows in result set
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/be2c2653728c52ac0d08c8972ba7c01d/logTestLargeUnsplittableRangeReplicate644991607
--- FAIL: TestLargeUnsplittableRangeReplicate (60.92s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLargeUnsplittableRangeReplicate.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-32586

kv/kvserver.TestLargeUnsplittableRangeReplicate [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/12358539?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/12358539?buildTab=artifacts#/) on master @ [c2ffc96f93f7a0dac9799788a042f962148e8af0](https://github.com/cockroachdb/cockroach/commits/c2ffc96f93f7a0dac9799788a042f962148e8af0):


```
=== RUN   TestLargeUnsplittableRangeReplicate
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/be2c2653728c52ac0d08c8972ba7c01d/logTestLargeUnsplittableRangeReplicate3640194246
    test_log_scope.go:81: use -show-logs to present logs inline
    replicate_queue_test.go:1869: replicas: {1,2,3}
    replicate_queue_test.go:1869: replicas: {1,2,3}
    replicate_queue_test.go:1869: replicas: {1,2,3}
    replicate_queue_test.go:1869: replicas: {1,2,3}
    replicate_queue_test.go:1869: replicas: {1,2,3}
    replicate_queue_test.go:1869: replicas: {1,2,3}
    replicate_queue_test.go:1869: replicas: {1,2,3}
    replicate_queue_test.go:1869: replicas: {1,2,3}
    replicate_queue_test.go:1869: replicas: {1,2,3}
    replicate_queue_test.go:1869: replicas: {1,2,3}
    replicate_queue_test.go:1869: replicas: {1,2,3}
    replicate_queue_test.go:1869: replicas: {1,2,3}
    replicate_queue_test.go:1869: replicas: {1,2,3}
    replicate_queue_test.go:1869: replicas: {1,2,3}
    replicate_queue_test.go:1869: replicas: {1,2,3}
    replicate_queue_test.go:1869: replicas: {1,2,3}
    replicate_queue_test.go:1869: replicas: {1,2,3}
    replicate_queue_test.go:1869: replicas: {1,2,3}
    replicate_queue_test.go:1869: replicas: {1,2,3}
    replicate_queue_test.go:1869: replicas: {1,2,3}
    replicate_queue_test.go:1869: replicas: {1,2,3}
    replicate_queue_test.go:1869: replicas: {1,2,3}
    replicate_queue_test.go:1869: replicas: {1,2,3}
    replicate_queue_test.go:1869: replicas: {1,2,3}
    replicate_queue_test.go:1869: replicas: {1,2,3}
    replicate_queue_test.go:1869: replicas: {1,2,3}
    replicate_queue_test.go:1869: replicas: {1,2,3}
    replicate_queue_test.go:1869: replicas: {1,2,3}
    replicate_queue_test.go:1869: replicas: {1,2,3}
    replicate_queue_test.go:1869: replicas: {1,2,3}
    replicate_queue_test.go:1869: replicas: {1,2,3}
    replicate_queue_test.go:1869: replicas: {1,2,3,4,5}
    replicate_queue_test.go:1877: condition failed to evaluate within 45s: sql: no rows in result set
    panic.go:523: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/be2c2653728c52ac0d08c8972ba7c01d/logTestLargeUnsplittableRangeReplicate3640194246
--- FAIL: TestLargeUnsplittableRangeReplicate (61.32s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLargeUnsplittableRangeReplicate.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/kv/kvserver/replicate_queue_test.go
+++ b/pkg/kv/kvserver/replicate_queue_test.go
@@ -1774,8 +1774,6 @@ func TestLargeUnsplittableRangeReplicate(t *testing.T) {
 		base.TestClusterArgs{
 			ReplicationMode: base.ReplicationAuto,
 			ServerArgs: base.TestServerArgs{
-				ScanMinIdleTime: time.Millisecond,
-				ScanMaxIdleTime: time.Millisecond,
 				Knobs: base.TestingKnobs{
 					Server: &server.TestingKnobs{
 						DefaultZoneConfigOverride: &zcfg,
@@ -1788,11 +1786,12 @@ func TestLargeUnsplittableRangeReplicate(t *testing.T) {
 
 	// We're going to create a table with many versions of a big row and a small
 	// row. We'll split the table in between the rows, to produce a large range
-	// and a small one. Then we'll increase the replication factor to 5 and check
-	// that both ranges behave the same - i.e. they both get up-replicated. For
-	// the purposes of this test we're only worried about the large one
-	// up-replicating, but we test the small one as a control so that we don't
-	// fool ourselves.
+	// and a small one. We'll also split the first row into its own range, to
+	// avoid the range inheriting 5 replicas from the system ranges. Then we'll
+	// increase the replication factor to 5 and check that both ranges behave the
+	// same - i.e. they both get up-replicated. For the purposes of this test
+	// we're only worried about the large one up-replicating, but we test the
+	// small one as a control so that we don't fool ourselves.
 
 	// Disable the queues so they don't mess with our manual relocation. We'll
 	// re-enable them later.
@@ -1804,6 +1803,8 @@ func TestLargeUnsplittableRangeReplicate(t *testing.T) {
 	require.NoError(t, err)
 
 	_, err = db.Exec(`ALTER TABLE t EXPERIMENTAL_RELOCATE VALUES (ARRAY[1,2,3], 1)`)
+	require.NoError(t, err)
+	_, err = db.Exec(`ALTER TABLE t SPLIT AT VALUES (1)`)
 	require.NoError(t, err)
 	_, err = db.Exec(`ALTER TABLE t SPLIT AT VALUES (2)`)
 	require.NoError(t, err)
@@ -1862,7 +1863,7 @@ func TestLargeUnsplittableRangeReplicate(t *testing.T) {
 	testutils.SucceedsSoon(t, func() error {
 		forceProcess()
 		r := db.QueryRow(
-			"SELECT replicas FROM [SHOW RANGES FROM TABLE t] WHERE start_key LIKE '%TableMin%'")
+			"SELECT replicas FROM [SHOW RANGES FROM TABLE t] WHERE start_key LIKE '%/1'")
 		var repl string
 		if err := r.Scan(&repl); err != nil {
 			return err

--- a/pkg/kv/kvserver/replicate_queue_test.go
+++ b/pkg/kv/kvserver/replicate_queue_test.go
@@ -1811,21 +1811,6 @@ func TestLargeUnsplittableRangeReplicate(t *testing.T) {
 	toggleReplicationQueues(tc, true /* active */)
 	toggleSplitQueues(tc, true /* active */)
 
-	// Check that the two ranges exist for table t.
-	testutils.SucceedsSoon(t, func() error {
-		r := db.QueryRow(
-			"SELECT count(*) FROM [SHOW RANGES FROM TABLE t]")
-		var count int
-		if err := r.Scan(&count); err != nil {
-			return err
-		}
-		if count != 2 {
-			return fmt.Errorf(
-				"splits not created, expected %d ranges, found %d", 2, count)
-		}
-		return nil
-	})
-
 	// We're going to create a large row, but now large enough that write
 	// back-pressuring kicks in and refuses it.
 	var sb strings.Builder

